eox_core/edxapp_wrapper/backends/users_h_v1.py

```diff
@@ -64,69 +66,70 @@ def create_edxapp_user(*args, **kwargs):
         return None, ["Fatal: account collition with the provided: {}".format(", ".join(conflicts))]
+    request = kwargs.pop("request", False)
+    request.data['name'] = kwargs.pop("fullname")
+    # Get additional settings for this user
+    email_validation = request.data.pop("verify_account_email", False)
+    terms_of_service = request.data.pop("terms_of_service", None)
+    honor_code = request.data.pop("honor_code", None)
+    comments_service = request.data.pop("enable_comments_service", None)
+    extended_profile_data = request.data.pop("extended_profile_fields", {})
+
+    # Settings to be monkey patched to suit request
+    registration_fields = getattr(settings, 'REGISTRATION_EXTRA_FIELDS', {})
+    site_configuration = configuration_helpers.get_current_site_configuration()
+    # Backup copies of original configurations
+    _original_settings_features = deepcopy(settings.FEATURES)
+    _original_settings_registration_fields = deepcopy(registration_fields)
+
+    settings.FEATURES['SKIP_EMAIL_VALIDATION'] = not email_validation
+
+    if not comments_service:
+        if 'ENABLE_DISCUSSION_SERVICE' in settings.FEATURES:
+            settings.FEATURES['ENABLE_DISCUSSION_SERVICE'] = False
+
+    if terms_of_service:
+        registration_fields['terms_of_service'] = 'required'
+    else:
+        registration_fields['terms_of_service'] = 'hidden'
 
-    data = {
-        'username': username,
-        'email': email,
-        'password': kwargs.pop("password"),
-        'name': kwargs.pop("fullname"),
-    }
-    # Go ahead and create the new user
-    with transaction.atomic():
-        # In theory is possible to extend the registration form with a custom app
-        # An example form app for this can be found at http://github.com/open-craft/custom-form-app
-        # form = get_registration_extension_form(data=params)
-        # if not form:
-        form = AccountCreationForm(
-            data=data,
-            tos_required=False,
-            # TODO: we need to support the extra profile fields as defined in the django.settings
-            # extra_fields=extra_fields,
-            # extended_profile_fields=extended_profile_fields,
-            # enforce_password_policy=enforce_password_policy,
-        )
-        (user, profile, registration) = do_create_account(form)  # pylint: disable=unused-variable
-
-    site = kwargs.pop("site", False)
-    if site:
-        create_or_set_user_attribute_created_on_site(user, site)
+    if honor_code:
+        registration_fields['honor_code'] = 'required'
     else:
-        errors.append("The user was not assigned to any site")
+        registration_fields['honor_code'] = 'hidden'
 
-    try:
-        create_comments_service_user(user)
-    except Exception:  # pylint: disable=broad-except
-        errors.append("No comments_service_user was created")
-
-    # TODO: link account with third party auth
-
-    lang_pref = kwargs.pop("language_preference", False)
-    if lang_pref:
```

Comment:
Can you remove a block like this? We still want to support setting the `language_preference` so unless that feature is supported on create_account_with_params then we need to keep it.